### PR TITLE
Fix for failing test: test_connection_error

### PR DESCRIPTION
**Pull request by colindickson**
http://fooobarbangbazbing.httpbin.org does not seem to be an unknown domain anymore.  I've also split up this unit test into two tests since they test slightly different things.

### CODE REVIEW

File: test_requests.py [hunk 1]
```diff
@@ -301,11 +301,13 @@ def test_BASICAUTH_TUPLE_HTTP_200_OK_GET(self):
...
+    def test_connection_error_invalid_domain(self):
         """Connecting to an unknown domain should raise a ConnectionError"""
         with pytest.raises(ConnectionError):
-            requests.get("http://fooobarbangbazbing.httpbin.org")
+            requests.get("http://doesnotexist.google.com")
 
+    def test_connection_error_invalid_port(self):
+        """Connecting to an invalid port should raise a ConnectionError"""
         with pytest.raises(ConnectionError):
             requests.get("http://httpbin.org:1")
```
Comment (kevinburke) [inline test_requests.py:312]: Huh. This line hangs on my machine and I'm pretty sure I wrote this test. Maybe your local machine is sending back a TCP RST, mine isn't

**Comment by kevinburke** [comment]
LGTM

File: test_requests.py (hunk 1 above)
Comment (kevinburke) [inline test_requests.py:312]: Passes on ci.kennethreitz.org anyway

**Comment by Lukasa** [comment]
Thanks! :cake:

File: test_requests.py (hunk 1 above)
Comment (colindickson) [inline test_requests.py:312]: @kevinburke i will make another PR today which will fix the hang.   (i didn't notice the hang last night on my windows machine, but i do have it on my ubuntu machine.)   cheers.

**Pull request merged**